tests/backup/backup_test.go

```diff
@@ -86,6 +88,508 @@ var _ = Describe("{UserGroupManagement}", func() {
+					resp, err := backupDriver.InspectRestore(ctx, restoreInspectRequest)
+					log.FailOnError(err, "Failed to inspect restore: %s", initialRestoreName)
+
+					log.InfoD("Switching to destination context")
+					err = SetDestinationKubeConfig()
+					log.FailOnError(err, "Failed to switch to context to destination cluster")
+
+					rstCtx, err := ValidateRestore(resp, backupContexts[i], namespaceMapping)
+
+					log.InfoD("Switching to default context")
+					err1 := SetClusterContext("")
+					log.FailOnError(err1, "Failed to SetClusterContext to default cluster")
+
+					dash.VerifyFatal(err, nil, fmt.Sprintf("Restore 1 (%s) Validation", initialRestoreName))
+
+					restoreContexts = append(restoreContexts, rstCtx)
+				})
+
+			}
+		})
+
+	})
+
+	JustAfterEach(func() {
+
+		defer EndTorpedoTest()
+
+		ctx, err := backup.GetAdminCtxFromSecret()
+		log.FailOnError(err, "Fetching px-central-admin ctx")
+
+		// TODO: move this to AfterSuite
+		if len(preRuleNameList) > 0 {
+			for _, ruleName := range preRuleNameList {
+				err := Inst().Backup.DeleteRuleForBackup(orgID, ruleName)
+				dash.VerifySafely(err, nil, fmt.Sprintf("Deleting backup pre rules %s", ruleName))
+			}
+		}
+
+		// TODO: move this to AfterSuite
+		if len(postRuleNameList) > 0 {
+			for _, ruleName := range postRuleNameList {
+				err := Inst().Backup.DeleteRuleForBackup(orgID, ruleName)
+				dash.VerifySafely(err, nil, fmt.Sprintf("Deleting backup post rules %s", ruleName))
+			}
+		}
+
+		opts := make(map[string]bool)
+		opts[SkipClusterScopedObjects] = false
+		log.InfoD("Deleting deployed applications for source and destintion clusters")
+
+		log.InfoD("Switching to source context")
+		err = SetSourceKubeConfig()
+		log.FailOnError(err, "Failed to switch to context to source cluster")
+
+		log.Info("Deleting deployed applications on source clusters")
+		ValidateAndDestroy(sourceClusterAppsContexts, opts)
+
+		log.Warn("Waiting for 1 minute, hoping for any Resources created by Operator of Custom Resources to finish being destroyed.\nThis logic will have to be changed to polling in the future, as just randomly waiting, without verification can easily lead to errors. Make sure that any errors you're seeing is not due to this.")
+		time.Sleep(time.Minute * 1)
+
+		log.InfoD("Switching to destination context")
+		err = SetDestinationKubeConfig()
+		log.FailOnError(err, "Failed to switch to context to destination cluster")
+
+		destinationClusterAppsContexts = make([]*scheduler.Context, 0)
+		// only adding restoreContexts, not restoreLaterContexts
+		for _, rstCtx := range restoreContexts {
+			destinationClusterAppsContexts = append(destinationClusterAppsContexts, rstCtx.schedCtxs...)
+		}
+		log.Info("Deleting deployed applications (Restore 1) on destintion clusters")
+		ValidateAndDestroy(destinationClusterAppsContexts, opts)
+
+		//TODO: this can be added in the future
+		//log.Warn("Not waiting here since we're destroying two namespaces, so we can wait 'at once'")
```

Comment:
Why do you need this commented code?

Reply:
I modified it to simply warn that the restore-later namespace isn't cleaned up